docs/auth/identity-resolver.md

```diff
@@ -90,116 +83,156 @@ the given auth provider, as well as a context object that contains various helpe
+```ts title="in packages/backend/src/index.ts"
+/* highlight-add-start */
+import { createBackendModule } from '@backstage/backend-plugin-api';
+import { githubAuthenticator } from '@backstage/plugin-auth-backend-module-github-provider';
+import {
+  authProvidersExtensionPoint,
+  createOAuthProviderFactory,
+} from '@backstage/plugin-auth-node';
+
+const customAuth = createBackendModule({
+  // This ID must be exactly "auth" because that's the plugin it targets
+  pluginId: 'auth',
+  // This ID must be unique, but can be anything
+  moduleId: 'custom-auth-provider',
+  register(reg) {
+    reg.registerInit({
+      deps: { providers: authProvidersExtensionPoint },
+      async init({ providers }) {
+        providers.registerProvider({
+          // This ID must match the actual provider config, e.g. addressing
+          // auth.providers.github means that this must be "github".
+          providerId: 'github',
+          // Use createProxyAuthProviderFactory instead if it's one of the proxy
+          // based providers rather than an OAuth based one
+          factory: createOAuthProviderFactory({
+            authenticator: githubAuthenticator,
+            async signInResolver(info, ctx) {
+              /*********************************************************************
+               * Custom resolver code goes here, see farther down in this article! *
+               * "info" is the sign in result from the upstream (github here), and *
+               * "ctx" contains useful utilities for token issuance etc.           *
+               *********************************************************************/
+            },
+          }),
+        });
+      },
+    });
+  },
+});
+/* highlight-add-end */
 
-```ts
-// File: packages/backend/src/plugins/auth.ts
-export default async function createPlugin(
-  // ...
-  return await createRouter({
-    // ...
-    providerFactories: {
-      // ...
-      google: providers.google.create({
-        signIn: {
-          resolver: providers.google.resolvers.emailLocalPartMatchingUserEntityName(),
-        },
-      });
-    }
-  })
-)
+backend.add(import('@backstage/plugin-auth-backend'));
+/* highlight-remove-next-line */
+backend.add(import('@backstage/plugin-auth-backend-module-github-provider'));
+/* highlight-add-next-line */
+backend.add(customAuth);
 ```
 
-There are also other options, like the this one that looks up a user
-by matching the `google.com/email` annotation of user entities in the catalog:
+The `createOAuthProviderFactory` / `createProxyAuthProviderFactory` functions
+have additional options for profile and state transforms - not covered here, but
+good to know about if you need them.
+
+So what would the body of a typical sign in resolver callback look like? Here's
+an example:
 
 ```ts
-providers.google.create({
-  signIn: {
-    resolver: providers.google.resolvers.emailMatchingUserEntityAnnotation(),
-  },
-});
+// ...
+async signInResolver(info, ctx) {
```

Comment:
Is it possible to define multiple resolvers? As mentioned in the Azure config example [here](https://github.com/backstage/backstage/pull/24154/files#diff-63897d51236aba0e4f41e7f390d1cc78aa592ebf42a0c8ff706958c484eb50cdR115-R119), right? I think it would me nice add one example doing that as well.

Reply:
no, if you do it in code you can only pass in one - but that's just because there's nothing stopping you from trying different things in code inside that single resolver, rather than the framework having to deal with a loop and corner cases